include/bitcoin/system/wallet/witness_address.hpp

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) 2011-2019 libbitcoin developers (see AUTHORS)
+ *
+ * This file is part of libbitcoin.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef LIBBITCOIN_SYSTEM_WALLET_WITNESS_ADDRESS_HPP
+#define LIBBITCOIN_SYSTEM_WALLET_WITNESS_ADDRESS_HPP
+
+#include <bitcoin/system/math/hash.hpp>
+#include <bitcoin/system/wallet/payment_address.hpp>
+
+namespace libbitcoin {
+namespace system {
+namespace wallet {
+
+/// A class for working with standard witness payment addresses.
+class BC_API witness_address
+  : public payment_address
+{
+  public:
+    enum class address_format: uint8_t
+    {
+        p2wpkh,
+        p2wsh
```

Comment:
Style: acronyms :/. Could you review against existing similar terms in `-system` and update for consistency?